triemap_alloc.h

```diff
@@ -0,0 +1,14 @@
+/* Allocator selection.
+ *
+ * This file is used in order to change the triemap allocator.
+ * You can define the following to your custom allocator, or leave as is for the libc default. */
+
+#ifndef __TRIEMAP_ALLOC_H__
+#define __TRIEMAP_ALLOC_H__
+
+#define triemap_malloc malloc
```

Comment:
this does not allow injecting allocators without actually changing the code in triemap. 
How about checking some ifdef here and only defining this if the allocators aren't already defined? That way you can include your own file before calling any triemap includes, and inject without changing the code. e.g.:

either:
```c

#ifndef TRIEMAP_CUSTOM_ALLOCATORS
#define triemap_malloc malloc
....
#endif
```

or:

```c
#ifndef triemap_malloc
#define triemap_malloc malloc
#endif
... and so forth
```

Option 2 is more elegant but it allows you to more easily fuck up... In option 1 if you forgot to set one of them the compilation will fail.